### PR TITLE
prov/sockets: Removed common code and added a function to get src_add

### PR DESCRIPTION
**Pull request by shantonu**
-Removed common code to get src_addr and created a function sock_get_src_addr_from_hostname()

@jithinjosepkl, can you please review?
Signed-off-by: Shantonu Hossain <shantonu.hossain@intel.com>

### CODE REVIEW

**Comment by jithinjosepkl** [comment]
Can we do the same for gethostname calls in sock_conn.c and sock_ep_msg.c?

**Comment by shantonu** [comment]
Sure! I'll update the PR.

**Comment by shantonu** [comment]
Updated the PR. Called <code>sock_get_src_addr_from_hostname()</code> from all the places that did the same thing. @jithinjosepkl, can you please review?

File: prov/sockets/src/sock_ep_msg.c [hunk 1]
```diff
@@ -326,24 +325,10 @@ static int sock_pep_create_listener(struct sock_pep *pep)
...
-		ret = getaddrinfo(hostname, sa_port, &hints, &s_res);
-		if (ret) {
-			SOCK_LOG_DBG("getaddrinfo failed!\n");
-			return -FI_EINVAL;
-		}
-		memcpy(&pep->src_addr, (struct sockaddr_in *)s_res->ai_addr,
-		       sizeof pep->src_addr);
-		freeaddrinfo(s_res);
+		ret = sock_get_src_addr_from_hostname(&pep->src_addr, &sa_port);
+		if (ret)
```
Comment (shantonu) [inline prov/sockets/src/sock_ep_msg.c]: Changed it to sa_port.

**Comment by jithinjosepkl** [comment]
Thanks, looks good.

**Pull request merged**